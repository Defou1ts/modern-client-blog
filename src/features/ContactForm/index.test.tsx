import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import { ContactForm } from './index';

describe('ContactForm', () => {
	it('renders a  ContactForm', () => {
		render(<ContactForm />);
		const text = screen.getByText('contact.form.related');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('ContactForm snapshot', () => {
		const component = renderer.create(<ContactForm />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
