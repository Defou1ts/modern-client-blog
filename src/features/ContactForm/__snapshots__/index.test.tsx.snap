// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm ContactForm snapshot 1`] = `
<section>
  <form
    action="#"
    className="form"
    data-test-id="contact-form"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="wrapper"
    >
      <input
        className="input input"
        data-test-id="contact-form-input-name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="contact.form.name"
        value=""
      />
    </div>
    <div
      className="wrapper"
    >
      <input
        className="input input"
        data-test-id="contact-form-input-email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="contact.form.email"
        value=""
      />
    </div>
    <div
      className="wrapper"
    >
      <select
        className="input select"
        data-test-id="contact-form-select"
        name="location"
        onBlur={[Function]}
        onChange={[Function]}
        value=""
      >
        <option
          value=""
        >
          contact.form.related
        </option>
        <option
          data-test-id="contact-form-option"
          value="Warsaw"
        >
          contact.form.locations.warsaw
        </option>
        <option
          data-test-id="contact-form-option"
          value="Minsk"
        >
          contact.form.locations.minsk
        </option>
        <option
          data-test-id="contact-form-option"
          value="London"
        >
          contact.form.locations.london
        </option>
      </select>
      <img
        alt="arrow-down"
        className="arrow"
        data-nimg="1"
        decoding="async"
        height={40}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={40}
      />
    </div>
    <div
      className="wrapper"
    >
      <textarea
        className="textarea textarea"
        data-test-id="contact-form-textarea"
        name="message"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="contact.form.message"
        value=""
      />
    </div>
    <button
      className="button primary"
      data-test-id="contact-form-submit"
      type="submit"
    >
      contact.form.send-button
    </button>
  </form>
</section>
`;
